### DISCUSSION TITLE
HTML exporting

### DISCUSSION THREAD

**Discussion by nachog00** [yoopta-editor/Yoopta-Editor#137] [Q&A]
Thanks for the great plugin! I'm surely be using it a lot.
Now my question is:
is there any way to export the content as a plain html? I intend on using this as a high quality input in a form, and I need to store the input as html.
I saw the editor.getEditorValue method returns an object with a tree but i cant really use that as it is.
Thanks anyways.

**Comment by codebychandu** [comment]
If you want to export the content as plain HTML while still using BeautifulSoup for parsing and manipulation, you can achieve this by converting the Markdown back to HTML after processing it with BeautifulSoup.from bs4 import BeautifulSoup
import markdown

def html_to_md(html_content):
    soup = BeautifulSoup(html_content, 'html.parser')

    md_content = markdown.markdown(str(soup))

    return md_content

def md_to_html(md_content):
    html_content = markdown.markdown(md_content)

    return html_content

def write_content_to_file(content, file_path):
    with open(file_path, 'w') as file:
        file.write(content)

def main():
    with open('input.html', 'r') as file:
        html_content = file.read()
    md_content = html_to_md(html_content)
    updated_html_content = md_to_html(md_content)
    write_content_to_file(updated_html_content, 'output.html')

if __name__ == "__main__":
    main()

**Comment by nachog00** [comment]
Hey tanks for your answer.
IM still not sure how this would be used. Should i pass the editor rendered html to it and i would get plain html as its seen?

**Comment by Darginec05** [comment]
Hi @nachog00. Good question.
For these purposes we have separate package `@yoopta/exports`.
It will support `html, md, mjml, text etc.` serialize and deserialize methdos. 
But this package should be upgraded from v2 to v4, because yoopta@v4 have completely different content structure

So, I'll convert this discussion into issue to finish `@yoopta/exports` this package :) 
Thanks!

**Reply by nachog00** [reply]
Oh I see. Thanks for the clarification.
In that case, I'll wait for the packages.
Only since I read on the main readme that html was already supported, would that be on V2?

**Comment by Darginec05** [comment]
Attention please! Good news is coming! 

I finally released the -rc version for @yoopta/exports 🔥
I also worked on examples to make it as convenient as possible for developers to understand how deserializers/serializers work. Check examples below 👇

So, what we have in current time: 
 - Version for packages - 4.5.2-rc.1
 - Full docs  - https://yoodocs.space/yoopta-editor/yoopta/exportsimports-fUZcaVYc7w
 - Examples - https://yoopta-editor-jt4typuh8-dargos-projects.vercel.app/examples/withExports
    - Example with HTML - https://yoopta-editor-jt4typuh8-dargos-projects.vercel.app/examples/withExports/html
    - Example with Markdown - https://yoopta-editor-jt4typuh8-dargos-projects.vercel.app/examples/withExports/markdown

But this task is still in progress.
What missing in current version:  - Cut content from editor
 - Copy/cut content from editor in readOnly mode
 - Parsing nested inline marks from markdown

I think these minor bugs will be fixed soon and I'll publish a stable version ASAP.
